### PR TITLE
pkgconfig: Use Libs.private for "-pthread" flag.

### LINKED ISSUE #21: Issue with Requires.private in pkgconfig file
**Describe the bug**
The `Requires.private` [line](https://github.com/jrmadsen/PTL/blob/6da7f0f999473cbb3c5af5594041ee885423c5b7/cmake/Templates/ptl.pc.in#L9) in the pkgconfig file seems to be erroneous according to [the pkg-config specifications](https://people.freedesktop.org/~dbn/pkg-config-guide.html). Both `Requires` and `Requires.private` should list names of pkg-config modules and not refer to flags directly, whereas `-pthread` looks more appropriate for `Libs.private`. Indeed on my system (openSUSE Tumbleweed) there does not seem to be any pkg-config module by the name of `-pthread` (or even `pthread`) installed even though PTL can find `pthread.h` when building and link against it just fine (`glibc-devel` provides the pthread header).

Please let me know if I am wrong here; if not, here is a straightforward patch which I can make a PR with if you wish:
```diff
Index: PTL-2.2.0/cmake/Templates/ptl.pc.in
===================================================================
--- PTL-2.2.0.orig/cmake/Templates/ptl.pc.in
+++ PTL-2.2.0/cmake/Templates/ptl.pc.in
@@ -6,6 +6,6 @@ Name: PTL
 Description: Parallel Tasking Library for C++
 Version: @PTL_VERSION_STRING@
 Requires: @PTL_PC_TBB_REQUIREMENT@
-Requires.private: -pthread
 Libs: -L${libdir} -lptl
+Libs.private: -pthread
 CFlags: -std=c++@CMAKE_CXX_STANDARD@ -pthread -I${includedir}
```

Many thanks for your library btw.

### PR DESCRIPTION
**Pull request by badshah400**
Fixes #21.

### CODE REVIEW

File: cmake/Templates/ptl.pc.in [hunk 1]
```diff
@@ -6,6 +6,6 @@ Name: PTL
 Description: Parallel Tasking Library for C++
 Version: @PTL_VERSION_STRING@
 Requires: @PTL_PC_TBB_REQUIREMENT@
-Requires.private: -pthread
+Libs.private: -pthread
 Libs: -L${libdir} -lptl
```
Comment (jrmadsen) [inline cmake/Templates/ptl.pc.in:10]:
```suggestion
Libs: -L${libdir} -lptl
Libs.private: -pthread
```

**Review by jrmadsen — COMMENTED** [review]
Just prefer this one change for readability

**Comment by jrmadsen** [comment]
@badshah400 Oh could you also bump the version (in VERSION) to 2.2.1

**Comment by badshah400** [comment]
> @badshah400 Oh could you also bump the version (in VERSION) to 2.2.1

Done. Many thanks for your quick responses and suggestions.

**Pull request merged**